### DISCUSSION TITLE
Why is `vim.g.persisting` not set on load ?

### DISCUSSION THREAD

**Discussion by dca123** [olimorris/persisted.nvim#101] [Q&A] [answered]
I installed the plugin with lazy as such.

```
	{
		"olimorris/persisted.nvim",
		config = true,
		lazy = true,
	},

```

Then in .config/nvim/after/plugin/persisted.lua, I loaded the telescope extension as such
```
require("telescope").load_extension("persisted")

vim.keymap.set("n", "<leader>ps", ":Telescope persisted<CR>", {})
```

My autosave (by default) feature was not working. The rest of the plugin worked perfectly as I could use `:SessionSave` and `:SessionLoad` to load up the sessions. After some investigating, I discovered that `vim.g.persisting` was not set. Manually setting this to true as shown below, enabled the autosave feature. 

```
require("telescope").load_extension("persisted")


vim.g.persisting = true
vim.keymap.set("n", "<leader>ps", ":Telescope persisted<CR>", {})
````

Any idea why this might be happening ?  

Cheers !

**Comment by olimorris** [comment, marked as answer]
It's because you've set `lazy = true`

**Comment by dca123** [comment]
Thank you for your response !

Tried it with removing `lazy = true` and then removing the `vim.g.persisting = true`.
Unfortunately the issue exists and vim confirms with `Undefined variable: g:persisting`

**Comment by dca123** [comment]
My assumption is that the session doesn't automatically start. After playing around a bit more, I noticed that running `:SessionStart` will setup the global variable and everything works as expected.

**Reply by olimorris** [reply]
So `lazy = true` will stop the plugin from starting so if you run `:lua print(vim.g.persisting)` you'll see `nil`. 

If you start looking at the code from [here](https://github.com/olimorris/persisted.nvim/blob/fc9f398393cc3bb0e4e81cb9f7c133cd2b21467f/lua/persisted/init.lua#L89) you'll see how this happens.

**Reply by dca123** [reply]
So I tried with both lazy = false, and lazy removed, both with `:lua print(vim.g.persisting)` resulting in `nil`.

I also looked into the function and manually added 

```
"olimorris/persisted.nvim",
config = true,
opts = {
			autosave = true,
			allowed_dirs = nil,
			ignored_dirs = nil,
		},
```
in hope that this might allow the function to succeed - alas no luck

**Reply by dca123** [reply]
Good news ! I had no idea what 'vim.fn.argc()' this does. Looked up the man files, and it's looking at the arguments. I've aliased my neovim as `n = nvim .` and it seems this is the culprit 🤣

All good now, working as intended ! Thanks for linking the code - I wouldn't have found this by myself 

Do you typically open up nvim with just 'nvim' as opposed to 'nvim .' ? I just switched from vscode a few days ago, and I used to do 'code .' to open up code in my cwd, so I just carried it over.

**Reply by olimorris** [reply]
Fantastic spot! That's one of those issues that could have taken ages to find 😄 .

My alias is `nv` and I have a Fish function:

```fish
function nv -d "Launch Neovim"
    if count $argv >/dev/null
        nvim $argv
    else
        nvim
    end
end
```

So it launches Neovim with `nv` and what ever other arguments I pass to it.

**Comment by RazaGR** [comment]
I encountered the same issue when I use `nvim .` (with dot) it doesn't autosave new changes, 
@olimorris is there any solution for that so we can autosave with `nvim .` `? 
🙏
